### PR TITLE
Add recursive navigation tutorial

### PR DESCRIPTION
**Pull request by excid3**
This is a 🔦 documentation change.

## Summary

This adds an example of using a recursive include to generate nested navigation links. I wrote about this on [my blog](http://excid3.com/blog/recursive-layout-with-jekyll-and-liquid) a long time ago and @arkadianriver mentioned we should add this to the docs in case it's helpful for other people. 👍

### CODE REVIEW

**Comment by ashmaroli** [comment]
```
    {% if item.tree %}
      {% include nav.html nav=item.subnav %}
    {% endif %}
```
Where is the `tree` attribute used above defined in?

**Comment by excid3** [comment]
@ashmaroli whoops, typo there! I figured `nav/subnav` would be easier to understand than my original code which used `tree`.

**Comment by excid3** [comment]
And fixed.

**Comment by ashmaroli** [comment]
I would suggest that you try to reproduce the tutorial step-by-step as written here, on a fresh boilerplate site. I still see some incoherent usage.

**Comment by excid3** [comment]
Fixed the `nav` and `navigation` mismatch and re-tested it. All is working nicely now.

nav

**Comment by ashmaroli** [comment]
Thanks for the quick fixes.

File: docs/_tutorials/navigation.md [hunk 1]
```diff
@@ -605,3 +605,65 @@ The `for item in items` loop looks through each `item` and gets the `title` and
 For more details on the `group_by` filter, see [Jekyll's Templates documentation](https://jekyllrb.com/docs/templates/) as well as [this Siteleaf tutorial](https://www.siteleaf.com/blog/advanced-liquid-group-by/). For more details on the `sort` filter, see [sort](https://shopify.github.io/liquid/filters/sort/) in Liquid's documentation.
 
 Whether you use properties in your doc's front matter to retrieve your pages or a YAML data file, in both cases you can programmatically build a more robust navigation for your site.
+
+## Scenario 9: Nested tree navigation with recursion
+
+Suppose you want a nested tree navigation of any depth. We can achieve this by recursively looping trough our tree of navigation links.
```
Comment (npras) [inline docs/_tutorials/navigation.md]: @excid3 typo here:

'looping trough'

Comment (excid3) [reply docs/_tutorials/navigation.md]: Thanks! Fixed.

**Comment by excid3** [comment]
Looks like Travis CI failed on an unrelated test. Is this normal?

**Comment by ashmaroli** [comment]
> Is this normal?

Don't worry about it.. It'll be handled..

**Comment by mattr-** [comment]
@excid3 Thanks so much for contributing this documentation!

@jekyllbot: merge +doc

**Pull request merged**